x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/top_metrics_agg/components/top_metrics_agg_form.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useContext } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiFormRow, EuiSelect, EuiButtonGroup } from '@elastic/eui';
+import { PivotAggsConfigTopMetrics } from '../types';
+import { PivotConfigurationContext } from '../../../../pivot_configuration/pivot_configuration';
+import {
+  isSpecialSortField,
+  KbnNumericType,
+  NUMERIC_TYPES_OPTIONS,
+  SORT_DIRECTION,
+  SORT_MODE,
+  SortDirection,
+  SortMode,
+  SortNumericFieldType,
+  TOP_METRICS_SORT_FIELD_TYPES,
+  TOP_METRICS_SPECIAL_SORT_FIELDS,
+} from '../../../../../../../common/pivot_aggs';
+
+export const TopMetricsAggForm: PivotAggsConfigTopMetrics['AggFormComponent'] = ({
+  onChange,
+  aggConfig,
+}) => {
+  const {
+    state: { fields },
+  } = useContext(PivotConfigurationContext)!;
+
+  const sortFieldOptions = fields
+    .filter((v) => TOP_METRICS_SORT_FIELD_TYPES.includes(v.type))
+    .map(({ name }) => ({ text: name, value: name }));
+
+  Object.values(TOP_METRICS_SPECIAL_SORT_FIELDS).forEach((v) => {
+    sortFieldOptions.unshift({ text: v, value: v });
+  });
+  sortFieldOptions.unshift({ text: '', value: '' });
+
+  const isSpecialFieldSelected = isSpecialSortField(aggConfig.sortField);
+
+  const sortDirectionOptions = Object.values(SORT_DIRECTION).map((v) => ({
+    id: v,
+    label: v,
```

Comment:
Should the labels in these options be internationalized? 🤔

Reply:
I don't recall seeing `asc` and `desc` to be translated in Kibana 🤔